src/Voron/Data/Compression/DecompressedLeafPage.cs

```diff
@@ -23,24 +22,18 @@ public DecompressedLeafPage(byte* basePtr, int pageSize, DecompressionUsage usag
         }
 
         public TreePage Original;
-        private readonly IDisposable _disposable;
+        private ByteStringContext.InternalScope _disposable;
 
         public bool Cached;
 
         public DecompressionUsage Usage;
 
         public void Dispose()
         {
-            // RavenDB-22090: We must not dispose leaf pages more than once.
-            if (_disposed)
```

Comment:
cannot happen now?

Reply:
It shouldn't NO. I have on my branch a commit that add those invariants as checks in debug too. But it was too much code to review on a single go :)

Reply:
Regardless, `_disposable.Dispose()` here is safe from calling this multiple times